packages/firebase_storage/lib/firebase_storage.dart

```diff
@@ -47,6 +47,11 @@ class StorageReference {
       },
     );
   }
+
+  Future<Null> deleteData(String path) {
```

Comment:
Just realized that this is an instance method on a `StorageReference` which already has a path and a way of getting to a child `StorageReference`, given a relative path.

I think we should change this new method to be just `delete()` without an argument. Then we can delete a child path by doing `myRef.child(path).delete();`.

Implementation methods across Dart, Java, and ObjC should then be changed from `deleteFile` to `delete`.

Sorry for not realizing this earlier.

Reply:
I also realised that after doing all the refactor. Initial idea was to wrap exception and give ability user to check if task was successful or not, but since we're now only executing it and catching exception, there's no point in having Task class and/or path in delete.